source/Detector/Amptek/AmptekSDD123Detector.cpp

```diff
@@ -0,0 +1,656 @@
+	tmpData[0] = tmpData[0] & 0x01;
+	internalSetClockAutoSet((bool)(tmpData.toHex().toInt(&ok, 16)));
+	//qDebug() << "clockAutoSet " << clockAutoSet_;
+
+	tmpData = statusData.mid(37, 1);
+	//UNUSED BYTE
+
+	tmpData = statusData.mid(38, 1);
+	tmpData[0] = tmpData[0] & 0x80;
+	tmpData[0] = tmpData[0] >> 7;
+	internalSetPc5DetectedAtPowerUp((bool)(tmpData.toHex().toInt(&ok, 16)));
+	//qDebug() << "pc5DetectedAtPowerUp " << pc5DetectedAtPowerUp;
+
+	tmpData = statusData.mid(36, 1);
+	tmpData[0] = tmpData[0] & 0x40;
+	tmpData[0] = tmpData[0] >> 6;
+	internalSetPc5HVNegative(!(bool)(tmpData.toHex().toInt(&ok, 16)));
+	internalSetPc5HVPositive((bool)(tmpData.toHex().toInt(&ok, 16)));
+	//qDebug() << "pc5HVNegative " << pc5HVNegative << " pc5HVPositive " << pc5HVPositive;
+
+	tmpData = statusData.mid(36, 1);
+	tmpData[0] = tmpData[0] & 0x20;
+	tmpData[0] = tmpData[0] >> 5;
+	internalSetPc5Preamp5V(!(bool)(tmpData.toHex().toInt(&ok, 16)));
+	internalSetPc5Preamp8p5V((bool)(tmpData.toHex().toInt(&ok, 16)));
+	//qDebug() << "pc5Preamp5V " << pc5Preamp5V << " pc5Preamp8p5V " << pc5Preamp8p5V;
+
+	tmpData = statusData.mid(39, 1);
+	internalSetDeviceID(tmpData.toHex().toInt(&ok, 16));
+	//qDebug() << "deviceID: " << deviceID;
+
+	//BYTES 40-63 ARE UNUSED
+	//qDebug() << "Emit statusDataRead" << name_;
+	emit statusDataRead();
+}
+
+void AmptekSDD123Detector::readTextConfigurationReadback(const QString &textConfiguration){
+	QStringList ASCIICommandList = textConfiguration.split(';', QString::SkipEmptyParts);
+	if(ASCIICommandList.contains("ALLP=??")){
+		ASCIICommandList.removeAll("ALLP=??");
+		configurationParametersInitialized_ = true;
+	}
+	QString ASCIICommand;
+	foreach(ASCIICommand, ASCIICommandList){
+		if(ASCIICommand.section('=', 0, 0) == "MCAC"){
+			//qDebug() << "I think MCAC is " << ASCIICommand.section('=', 1).toDouble();
+			//mcaChannels_->move(ASCIICommand.section('=', 1).toDouble());
+			internalSetMCAChannels(ASCIICommand.section('=', 1).toInt());
+		}
+		else if(ASCIICommand.section('=', 0, 0) == "PRET")
+			internalSetPresetTime(ASCIICommand.section('=', 1).toDouble());
+			//presetTime_->move(ASCIICommand.section('=', 1).toDouble());
+	}
+}
+
+void AmptekSDD123Detector::backwardsMid(int start, int length, const QByteArray &inputData, QByteArray &outputData){
+	outputData.clear();
+	for(int x = start+length; x > start; x--)
+		outputData.append(inputData.at(x-1));
+}
+
+QString AmptekSDD123Detector::stringFromBool(bool input) const{
+	return input ? "true" : "false" ;
+}
+
+
+void AmptekSDD123Detector::internalSetFastCounts(int fastCounts){
+	if(fastCounts != fastCounts_){
+		fastCounts_ = fastCounts;
+		emit fastCountsChanged(fastCounts_);
+		emit fastCountsChanged(QString("%1").arg(fastCounts_));
+	}
+}
+
+void AmptekSDD123Detector::internalSetSlowCounts(int slowCounts){
+	if(slowCounts != slowCounts_){
+		slowCounts_ = slowCounts;
+		emit slowCountsChanged(slowCounts_);
+		emit slowCountsChanged(QString("%1").arg(slowCounts_));
```

Comment:
Since this is going to be spitting out a lot of signals, would it make more sense to only emit the `int` versions of the signals and then have the specific slots that listen to it do the string conversion?  I think that doing a lot of string conversion all the time might be a bit wasteful.